servers/plugins/org.fusesource.ide.server.karaf.core/plugin.xml

```diff
@@ -346,32 +414,62 @@
+       			<facet id="jst.camel" version="3.0"/>
+     		</supported>
+     		<supported>
+        		<runtime-component
+         			id="org.fusesource.ide.karaf.runtime.component"
+         			version="4.0"/>
+                <facet id="wst.web"  version="1.0,1.2,1.3,1.4"/>
+       			<facet id="jst.web"  version="2.2,2.3,2.4,2.5,3.0"/>
+       			<facet id="jst.java" version="1.3,1.4,5.0,6.0,1.7,1.8"/>
+       			<facet id="jst.utility" version="1.0"/>
+       			<facet id="jst.camel" version="3.0"/>
+     		</supported>
+     		<supported>
+        		<runtime-component
+         			id="org.fusesource.ide.karaf.runtime.component"
+         			version="4.1"/>
+                <facet id="wst.web"  version="1.0,1.2,1.3,1.4"/>
+       			<facet id="jst.web"  version="2.2,2.3,2.4,2.5,3.0"/>
+       			<facet id="jst.java" version="1.3,1.4,5.0,6.0,1.7,1.8"/>
+       			<facet id="jst.utility" version="1.0"/>
+       			<facet id="jst.camel" version="3.0"/>
      		</supported>
    </extension>  
    
    <extension
          point="org.eclipse.wst.common.project.facet.core.runtimes">
-     <adapter>
+    <adapter>
       <runtime-component id="org.fusesource.ide.karaf.runtime.component" version="2.2"/>
       <factory class="org.fusesource.ide.server.karaf.core.runtime.classpath.KarafClasspathProviderAdapterFactory"/>
       <type class="org.eclipse.jst.common.project.facet.core.IClasspathProvider"/>
 	</adapter>
-     <adapter>
+    <adapter>
       <runtime-component id="org.fusesource.ide.karaf.runtime.component" version="2.3"/>
       <factory class="org.fusesource.ide.server.karaf.core.runtime.classpath.KarafClasspathProviderAdapterFactory"/>
       <type class="org.eclipse.jst.common.project.facet.core.IClasspathProvider"/>
 	</adapter>
-     <adapter>
+    <adapter>
       <runtime-component id="org.fusesource.ide.karaf.runtime.component" version="2.4"/>
       <factory class="org.fusesource.ide.server.karaf.core.runtime.classpath.KarafClasspathProviderAdapterFactory"/>
       <type class="org.eclipse.jst.common.project.facet.core.IClasspathProvider"/>
 	</adapter>
-     <adapter>
+    <adapter>
       <runtime-component id="org.fusesource.ide.karaf.runtime.component" version="3.0"/>
       <factory class="org.fusesource.ide.server.karaf.core.runtime.classpath.KarafClasspathProviderAdapterFactory"/>
       <type class="org.eclipse.jst.common.project.facet.core.IClasspathProvider"/>
 	</adapter>
+	<adapter>
+      <runtime-component id="org.fusesource.ide.karaf.runtime.component" version="4.0"/>
+      <factory class="org.fusesource.ide.server.karaf.core.runtime.classpath.KarafClasspathProviderAdapterFactory"/>
+      <type class="org.eclipse.jst.common.project.facet.core.IClasspathProvider"/>
+	</adapter>
+	<adapter>
+      <runtime-component id="org.fusesource.ide.karaf.runtime.component" version="4.1"/>
+      <factory class="org.fusesource.ide.server.karaf.core.runtime.classpath.KarafClasspathProviderAdapterFactory"/>
```

Comment:
as it is the same factory fo r4.0 and 4.1, shouldn't it be simple to have a single one for 4+ version?

Reply:
Yeah, good point. We should merge them into a single adapter 4+ for now.

Reply:
when looking at other server adapter plugins in JBT they also separate the versions. I'd like to keep it separated then.

Reply:
ok